pkg/distributor/distributor_retry.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributor
+
+import (
+	"flag"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	errNonPositiveRetryBase          = errors.New("Retry base duration should be a positive value")
+	errNonPositiveMaxAllowedAttempts = errors.New("MaxAllowedAttempts should be a positive value")
+)
+
+type RetryConfig struct {
+	Base    time.Duration `yaml:"base" category:"experimental"`
+	Enabled bool          `yaml:"enabled" category:"experimental"`
+	// MaxAllowedAttempts limits the number of retry attempts considered for calculation,
+	// capping it at the specified value. Additional attempts beyond this limit are not rejected.
+	// Used to control exponential backoff calculation with an upper limit.
+	MaxAllowedAttempts int `yaml:"max_allowed_attempts" category:"experimental"`
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet.
+func (cfg *RetryConfig) RegisterFlags(f *flag.FlagSet) {
+	f.DurationVar(&cfg.Base, "distributor.retry.base", 3*time.Second, "Base time to wait before retrying a 429/5xx request.")
```

Comment:
it would allow user to have an ability to change the steps of retry. as discussed most of user would go with default. will leave it like this for now.